.github/workflows/main.yml

```diff
@@ -102,20 +141,57 @@ jobs:
       # ensure that the tests fail more quickly so that people can fix them.
       timeout-minutes: 30
 
-    - name: Build executor image
-      run: ./tools/executor/build.js
+  native-checks:
+    runs-on: ubuntu-latest
 
-    # The following steps won't run correctly for forks since the Dockerhub
-    # credentials secrets will be missing. We only run them if the secrets are
-    # provided. Note that we can't use `secrets.*` directly in an `if` conditional
-    # here, so we copy them to an environment variable at the start of the job
-    # and then check that here.
-    - name: Login to DockerHub
-      uses: docker/login-action@v2
-      if: ${{ env.DOCKERHUB_USERNAME_SECRET != '' }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Install OS packages
+      run: sudo apt-get install -y graphviz graphviz-dev
```

Comment:
Needed for `pygraphviz` to install successfully.

Reply:
Not sure if it matters, but there's also a graphviz specific setup action.

Reply:
I'm happy to keep just this command here; seems simpler than pulling in another dependency that probably just runs these same commands under the hood.